hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -1148,7 +1149,8 @@ class Emit[C](
           val childShape = pndVal.shapes(cb)
           val childStrides = pndVal.strides(cb)
 
-          val dataArray = childPType.dataType.loadCheapPCode(cb, childPType.dataPArrayPointer(pndVal.tcode[Long]))
+          val pndAddr = SingleCodePCode.fromPCode(cb, pndVal, region)
+          val dataArray = childPType.dataType.loadCheapPCode(cb, childPType.dataPArrayPointer(pndAddr.code.asInstanceOf[Code[Long]]))
```

Comment:
Isn't doing `.code.asInstanceOf` effectively the same as doing `tcode`? What's the win here?

Reply:
The win is in `SingleCodePCode` which ensures the PCode has only one code. `tcode` is going away from `PCode` in order to remove the assumption that one PCode is one Code[_].

Reply:
Basically `SingleCodePCode` is the One True Way to get a `Code[_]` out of a `PCode`